base/loading.jl

```diff
@@ -53,6 +53,45 @@ function find_all_in_cache_path(mod::Symbol)
     paths
 end
 
+const r_compilable = r"^#\s*pragma\s+compile(\s+true)?\s*$"
+const r_ncompilable = r"^#\s*pragma\s+compile\s+false\s*$"
+
+# return true if "#pragma compilable [true]" appears in the file before
+# any Julia code, false for "#pragma compile false", default otherwise
+function compilable(path::AbstractString, default::Bool=false)
+    return open(path, "r") do f
+        for line in eachline(f)
+            s = lstrip(line)
+            if !isempty(s)
+                if s[1] == '#'
+                    ismatch(r_compilable, s) && return true
+                    ismatch(r_ncompilable, s) && return false
+                else
+                    return default
```

Comment:
I think this does the wrong thing for lines inside a block comment. Or `#pragma compilable` would also have to be documented as needing to come before any multi-line block comments.